src/ui/common/src/components/integrations/dialogs/s3Dialog.tsx

```diff
@@ -193,6 +194,19 @@ export const S3Dialog: React.FC<Props> = ({
         disableReason={editMode ? readOnlyFieldDisableReason : undefined}
       />
 
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="Directory"
+        description="An optional path to a directory in the S3 bucket to be used as the root. Defaults to the bucket's root."
```

Comment:
Let's make a note here that clarifies this is only relevant when used as an artifact store, not relevant when used as a regular data integration.